core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowVersionResource.scala

```diff
@@ -43,15 +137,32 @@ object WorkflowVersionResource {
       workflow.setCreationTime(patch.creationTime)
       workflow.setLastModifiedTime(patch.creationTime)
     }
-    // the checked out version is persisted to disk
+    // the checked out version is returned
     workflow
   }
-  case class VersionEntry(
+
+  trait VersionEntryDefinition {
+    def vId: UInteger
+    def creationTime: Timestamp
+    def content: String
+  }
+
+  case class VersionEntry(vId: UInteger, creationTime: Timestamp, content: String)
+      extends VersionEntryDefinition
+
+  /**
+    * This class is to add version importance encoding to the existing `VersionEntry` from DB
+    * @param vId
+    * @param creationTime
+    * @param content
+    * @param importance false is not an important version and true is an important version
+    */
+  case class ImpEncodedVersionEntry(
       vId: UInteger,
       creationTime: Timestamp,
-      content: String
-  )
-
+      content: String,
+      importance: Boolean
+  ) extends VersionEntryDefinition
 }
```

Comment:
ok so we have a `VersionEntry` and `ImpEncodedVersionEntry`. which are two case classes extended from a trait. 

This semantically seems wired to me, since `ImpEncodedVersionEntry` is a super set of `VersionEntry`.

how about: 
1. we can just add the importance field into `VersionEntry`. 

OR
2. or we keep tuples of `(versionEntry, importance)`

The concern is this `ImpEncodedVersionEntry` is not very extensive.

Reply:
I mentioned in the comments that `VersionEntry` can't directly have `importance` if I want to quickly import the DB entries result into the class. So this way is safer to always have `VersionEntry` matching the DB entries and `ImpEncodedVersionEntry` to be extendable so that in the future to include any extra columns not matching DB.

Reply:
yes understood. we have similar need in `WorkflowResource`, where `Workflow` is generated by jooq matching with db, however we need to keep some other information with it. Thus we created `DashboardWorkflowEntry`, which contains workflow, and other needed fields. Note that you are using a customized defined `VersionEntry` (the definition is not from jooq/db) already, why do you need another `ImpEncodedVersionEntry`?

Reply:
see if you can use jooq generated `WorkflowVersion` to replace `VersionEntry` and use `VersionEntry` to contain extra fields (like importance)?

Reply:
Makes sense. I followed the suggestion in a new commit.